frontend/src/views/CellGuide/components/CellGuideCard/components/MarkerGeneTables/index.tsx

```diff
@@ -191,234 +193,362 @@ export const MARKER_GENES_COMPUTATIONAL_TOOLTIP_TEST_ID =
+  setGeneInfoGene,
+}: {
+  genes: EnrichedGenesQueryResponse;
+  selectedOrganism: string;
+  selectedOrgan: string;
+  setGeneInfoGene: Props["setGeneInfoGene"];
+}): {
+  selectedOrganismFilter: string;
+  selectedOrganFilter: string;
+  tableRows: TableRow[];
+  uniqueOrganisms: string[];
+  uniqueOrgans: string[];
+} {
+  return useMemo(() => {
+    if (!genes)
+      return {
+        selectedOrganismFilter: selectedOrganism,
+        selectedOrganFilter: selectedOrgan,
+        tableRows: [],
+        uniqueOrganisms: [],
+        uniqueOrgans: [],
+      };
+
+    const organisms = new Set<string>();
+    for (const markerGene of genes) {
+      if (markerGene.marker_score < FMG_GENE_STRENGTH_THRESHOLD) continue;
+      organisms.add(markerGene.organism);
+    }
+
+    const sortedOrganisms = Array.from(organisms).sort((a, b) => {
+      if (a === "Homo sapiens") return -1;
+      if (b === "Homo sapiens") return 1;
+      return a.localeCompare(b);
+    });
+
+    const selectedOrganismFilter =
+      selectedOrganism === "" || !sortedOrganisms.includes(selectedOrganism)
+        ? sortedOrganisms.at(0) ?? ""
+        : selectedOrganism;
+
+    const organs = new Set<string>();
+    for (const markerGene of genes) {
+      if (markerGene.organism !== selectedOrganismFilter) continue;
+      if (markerGene.marker_score < FMG_GENE_STRENGTH_THRESHOLD) continue;
+      organs.add(markerGene.tissue);
+    }
+
+    const sortedOrgans = Array.from(organs).sort((a, b) => {
+      if (a === "All Tissues") return -1;
+      if (b === "All Tissues") return 1;
+      return a.localeCompare(b);
+    });
+
+    const selectedOrganFilter =
+      selectedOrgan === "" || !sortedOrgans.includes(selectedOrgan)
+        ? sortedOrgans.at(0) ?? ""
+        : selectedOrgan;
+
+    const rows: TableRow[] = [];
+    for (const markerGene of genes) {
+      const { pc, me, name, symbol, organism, marker_score, tissue } =
+        markerGene;
+      if (organism !== selectedOrganismFilter) continue;
+      if (tissue !== selectedOrganFilter) continue;
+      if (marker_score < FMG_GENE_STRENGTH_THRESHOLD) continue;
+      rows.push({
+        symbolId: symbol,
+        symbol: (
+          <>
+            {symbol}{" "}
+            <ButtonIcon
+              aria-label={`display gene info for ${symbol}`}
+              sdsIcon="infoCircle"
+              sdsSize="small"
+              sdsType="secondary"
+              onClick={() => setGeneInfoGene(symbol.toUpperCase())}
+            />
+          </>
+        ),
+        name,
+        marker_score: marker_score.toFixed(2),
+        me: me.toFixed(2),
+        pc: (pc * 100).toFixed(1),
+      });
+    }
+
+    return {
+      selectedOrganismFilter,
+      selectedOrganFilter,
+      tableRows: rows,
+      uniqueOrganisms: sortedOrganisms,
+      uniqueOrgans: sortedOrgans,
+    };
+  }, [genes, selectedOrganism, selectedOrgan, setGeneInfoGene]);
+}
+
+function getCanonicalMarkerGenesTableRowsAndFilters({
```

Comment:
refactored logic that existed in the main body of the functional component into this function.